docs/src/main/tut/git_data.md

```diff
@@ -0,0 +1,295 @@
+For example, the call to get the data for `master` branch will be `heads/master`.
+
+If the `ref` doesn't exist in the repository, but existing `refs` start with `ref` they will be
+returned as an array. For example, a call to get the data for a branch named `feature`,
+which doesn't exist, would return head refs including `featureA` and `featureB` which do.
+
+You can get a reference using `getReference`, it takes as arguments:
+
+- the repository coordinates (`owner` and `name` of the repository)
+- `ref`: ref formatted as heads/branch
+
+```tut:silent
+val getReference = Github(accessToken).gitData.getReference("47deg", "github4s", "heads/master")
+
+getReference.exec[cats.Id, HttpResponse[String]]() match {
+  case Left(e) => println("Something went wrong: s{e.getMessage}")
+  case Right(r) => println(r.result)
+}
+```
+
+The `result` on the right is the corresponding [List[Ref]][gitdata-scala].
+
+See [the API doc](https://developer.github.com/v3/git/refs/#get-a-reference) for full reference.
+
+
+### Create a Reference
+
+The ref in the URL must be formatted as `heads/branch`, not just branch.
+For example, the call to get the data for `master` branch will be `heads/master`.
+
+You can create a reference using `createReference`, it takes as arguments:
+
+- the repository coordinates (`owner` and `name` of the repository)
+- `ref`: The name of the fully qualified reference (ie: refs/heads/master).
+If it doesn't start with 'refs' and have at least two slashes, it will be rejected.
+- `sha`: the SHA1 value to set this reference to
+
+```scala
+val createReference = Github(accessToken)
+.gitData.createReference(
+                      "47deg",
+                      "github4s",
+                      "refs/heads/master",
+                      "d3b048c1f500ee5450e5d7b3d1921ed3e7645891")
+
+createReference.exec[cats.Id, HttpResponse[String]]() match {
+  case Left(e) => println("Something went wrong: s{e.getMessage}")
+  case Right(r) => println(r.result)
+}
+```
+
+The `result` on the right is the created [Ref][gitdata-scala].
+
+See [the API doc](https://developer.github.com/v3/git/refs/#create-a-reference) for full reference.
+
+
+### Update a Reference
+
+You can update a reference using `updateReference`, it takes as arguments:
+
+- the repository coordinates (`owner` and `name` of the repository)
+- `ref`: ref formatted as heads/branch
+- `sha`: the SHA1 value to set this reference to
+- `force`: Indicates whether to force the update or to make sure the update is a fast-forward update.
+Leaving this out or setting it to `false` will make sure you're not overwriting work. Default: `false`
+
+```scala
+val updateReference = Github(accessToken)
+.gitData.updateReference(
+                      "47deg",
+                      "github4s",
+                      "heads/master",
+                      "d3b048c1f500ee5450e5d7b3d1921ed3e7645891",
+                      Some(false))
```

Comment:
Also the `Option[Boolean]` seems weird, but that's not really related to this PR